datadog/fwprovider/resource_datadog_csm_threats_agent_rule.go

```diff
@@ -0,0 +1,218 @@
+			response.State.RemoveResource(ctx)
+			return
+		}
+		response.Diagnostics.Append(utils.FrameworkErrorDiag(err, "error fetching agent rule"))
+		return
+	}
+	if err := utils.CheckForUnparsed(response); err != nil {
+		response.Diagnostics.Append(utils.FrameworkErrorDiag(err, "response contains unparsed object"))
+		return
+	}
+
+	r.updateStateFromResponse(ctx, &state, &res)
+	response.Diagnostics.Append(response.State.Set(ctx, &state)...)
+}
+
+func (r *csmThreatsAgentRuleResource) Update(ctx context.Context, request resource.UpdateRequest, response *resource.UpdateResponse) {
+	var state csmThreatsAgentRuleModel
+	response.Diagnostics.Append(request.Plan.Get(ctx, &state)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	agentRulePayload, err := r.buildUpdateCSMThreatsAgentRulePayload(&state)
+	if err != nil {
+		response.Diagnostics.AddError("error while parsing resource", err.Error())
+	}
+
+	res, _, err := r.api.UpdateCSMThreatsAgentRule(r.auth, state.Id.ValueString(), *agentRulePayload)
+	if err != nil {
+		response.Diagnostics.Append(utils.FrameworkErrorDiag(err, "error updating agent rule"))
+		return
+	}
+	if err := utils.CheckForUnparsed(response); err != nil {
+		response.Diagnostics.Append(utils.FrameworkErrorDiag(err, "response contains unparsed object"))
+		return
+	}
+
+	r.updateStateFromResponse(ctx, &state, &res)
+	response.Diagnostics.Append(response.State.Set(ctx, &state)...)
+}
+
+func (r *csmThreatsAgentRuleResource) Delete(ctx context.Context, request resource.DeleteRequest, response *resource.DeleteResponse) {
+	var state csmThreatsAgentRuleModel
+	response.Diagnostics.Append(request.State.Get(ctx, &state)...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	id := state.Id.ValueString()
+
+	httpResp, err := r.api.DeleteCSMThreatsAgentRule(r.auth, id)
+	if err != nil {
+		if httpResp != nil && httpResp.StatusCode == 404 {
+			return
+		}
+		response.Diagnostics.Append(utils.FrameworkErrorDiag(err, "error deleting agent rule"))
+		return
+	}
+}
+
+func (r *csmThreatsAgentRuleResource) buildCreateCSMThreatsAgentRulePayload(state *csmThreatsAgentRuleModel) (*datadogV2.CloudWorkloadSecurityAgentRuleCreateRequest, error) {
+	_, name, description, enabled, expression := r.extractAgentRuleAttributesFromResource(state)
+
+	attributes := datadogV2.CloudWorkloadSecurityAgentRuleCreateAttributes{}
+	attributes.Expression = expression
+	attributes.Name = name
+	attributes.Description = description
+	attributes.Enabled = &enabled
+
+	data := datadogV2.NewCloudWorkloadSecurityAgentRuleCreateData(attributes, datadogV2.CLOUDWORKLOADSECURITYAGENTRULETYPE_AGENT_RULE)
+	return datadogV2.NewCloudWorkloadSecurityAgentRuleCreateRequest(*data), nil
+}
+
+func (r *csmThreatsAgentRuleResource) buildUpdateCSMThreatsAgentRulePayload(state *csmThreatsAgentRuleModel) (*datadogV2.CloudWorkloadSecurityAgentRuleUpdateRequest, error) {
+	agentRuleId, _, description, enabled, _ := r.extractAgentRuleAttributesFromResource(state)
```

Comment:
Since the name and expression cannot be updates, lets mark these fields in the schema with `RequiresReplace`